test/mciproject.yml

```diff
@@ -58,69 +83,86 @@ tasks:
+
+            ${test_env_vars|} TESTUTIL_ENABLE_LONG_TESTS=1 ${make_command|make} VERBOSE=1 check ${smp_command|} 2>&1
+
+  - name: examples-c-test
+    depends_on:
+      - name: compile
+    commands:
+      - func: "fetch artifacts"
+      - func: "compile wiredtiger"
+      - command: shell.exec
         params:
-          target: "wiredtiger.tgz"
-          source_dir: "wiredtiger"
-          include:
-             - "./**"
-      - command: s3.put
+          working_dir: "wiredtiger"
+          script: |
+            set -o errexit
+            set -o verbose
+
+            ${test_env_vars|} ${make_command|make} VERBOSE=1 check -C examples/c  ${smp_command|} 2>&1
+
+  - name: csuite-col-append-test
+    depends_on:
+      - name: compile
+    commands:
+      - func: "fetch artifacts"
+      - func: "compile wiredtiger"
+      - command: shell.exec
         params:
-          aws_secret: ${aws_secret}
-          aws_key: ${aws_key}
-          local_file: wiredtiger.tgz
-          bucket: build_external
-          permissions: public-read
-          content_type: application/tar
-          display_name: Binaries
-          remote_file: wiredtiger/${build_variant}/${revision}/binaries/${build_id}.tgz
+          working_dir: "wiredtiger"
+          script: |
+            set -o errexit
+            set -o verbose
+
+            ${test_env_vars|} TESTUTIL_ENABLE_LONG_TESTS=1 $(pwd)/test/csuite/test_wt2246_col_append 2>&1
 
   - name: unit-test
     depends_on:
     - name: compile
     commands:
-      - func: "fetch binaries"
+      - func: "fetch artifacts"
       - command: shell.exec
         params:
           working_dir: "wiredtiger"
           script: |
             set -o errexit
             set -o verbose
 
-            # Avoid /usr/bin/python, at least on macOS: with System Integrity
-            # Protection enabled, it ignores DYLD_LIBRARY_PATH and hence
-            # doesn't find the WiredTiger library in the local tree.
-            ${test_env_vars|} python ./test/suite/run.py -v 2 ${smp_command|} 2>&1
+            if [ "Windows_NT" = "$OS" ]; then
+              ${test_env_vars|} python ./test/suite/run.py -v 2 ${smp_command|} 2>&1
+            elif [ "$(uname -s)" == "Darwin" ]; then
+              # Avoid /usr/bin/python, at least on macOS: with System Integrity
+              # Protection enabled, it ignores DYLD_LIBRARY_PATH and hence
+              # doesn't find the WiredTiger library in the local tree.
+              ${test_env_vars|} python ./test/suite/run.py -v 2 ${smp_command|} 2>&1
+            else # Ubuntu
+              # Change directory to where the local installed 'wt' binary is located, 
+              # to avoid libtool generated 'wt' script from being selected by run.py,
+              # which invokes relink_command that tries to changing to a non-existed 
+              # /data/mci/<uniq> directory, as 'make' is done by a separate 'compile' task.
+              cd LOCAL_INSTALL/bin
```

Comment:
Seems like this `cd` is the primary difference - could you do this the other way, and `if #Ubuntu` do the `cd`, and then all other places run the same python command?

I see that the following line uses `test_env_vars_py` rather than `test_env_vars` - that distinction confuses me - I feel like it should either be made the same as other cases or renamed to make the difference more obvious

Reply:
`cd` here is critical here to make sure `run.py` locates the `wt` binary (instead of the `wt` script) to avoid non-existed directory error. I can change the checking sequence as suggestion. 

The `test_env_vars_py` addition is a bit ugly. It is needed for python test suite as the relative directory (to `$(pwd)`) becomes different after the `cd` command. Does `python_test_env_vars` make more sense?